app/forms/declaration/Document.scala

```diff
@@ -52,27 +49,30 @@ object Document extends DeclarationPage {
         "declaration.previousDocuments.documentReference.error",
         isEmpty or (isAlphanumericWithSpecialCharacters(Set('-', '/', ':')) and noLongerThan(35))
       ),
+    "documentCategory" -> requiredRadio("declaration.previousDocuments.documentCategory.error.empty")
+      .verifying("declaration.previousDocuments.documentCategory.error.empty", nonEmpty)
+      .verifying("declaration.previousDocuments.documentCategory.error.incorrect", isEmpty or isContainedIn(correctDocumentCategories))
+      .transform[DocumentCategory]({
+        case SimplifiedDeclaration.value => SimplifiedDeclaration
+        case RelatedDocument.value       => RelatedDocument
+      }, category => category.value),
```

Comment:
👍 
Nit: I've noticed that the `mapping` gets a bit long here. I don't remember the exact place, but we've had somewhere an example of having single field mappings assigned to values and then used these values in `mapping`.